### PR TITLE
Fit image options

### PR DESCRIPTION
**Pull request by Hiers**
Adding the image fit feature mentioned in #147.

The setFit function is hardcoded to look for the css file and attribute on those exact spots and looks likely to break if that file is changed. Asides from this, the only other way I know is to make a function to look for it given the name of the file and attribute, but it'd be a function that would only be used on this one occasion.

And if anyone has a better idea for how the option should appear in the settings I'd like to hear it, design is not my passion.

### CODE REVIEW

**Comment by Hiers** [comment]
Note: the real size option has the added benefit of having the browser's native scaling work as expected.

**Comment by Leeingnyo** [comment]
Thank you for PR!

> The setFit function is hardcoded to look for the css file and attribute on those exact spots and looks likely to break if that file is changed. Asides from this, the only other way I know is to make a function to look for it given the name of the file and attribute, but it'd be a function that would only be used on this one occasion.

We would write styles to classes (such as `.vertical`, `.horizontal`, `.original`), and in JS, adding classes to tags or removing classes from tags according to option value. I saw just codes and didn't see actual implementation on a browser yet. I'll tell you what would be a detailed way (which tags to have classes, what contents that class style would have) after seeing this on a browser.

**Comment by hkalexling** [comment]
Thanks! Yeah agreed with @Leeingnyo I think accessing the styles directly is a bit too hacky. I took a quick look and I think in the `setFit` function we can simply do `this.fitType = fitType` without touching the styles? The `img` style will be updated in `src/views/reader.html.ecr`.

**Comment by Leeingnyo** [comment]
@Hiers Sorry to late!

I misunderstood something. please forget about what i said before.

It seemed you used the hardcoded style to remove this following rule

```
canvas,
img,
video {
  /* 1 */
  max-width: 100%;
  /* 2 */
  height: auto;
  /* 3 */
  box-sizing: border-box;
}
```

Rather than removing to use `document.styleSheets[0].rules[21].style.maxWidth` directly, We could override a css rule like this

```html
<uk-img
  style="
    max-width:${fitType === 'horz' ? '100%' : 'none'}; /* <-- use 'none' instead of '' */
    max-height:${fitType === 'vert' ? '100%' : 'none'};
  "
/>
```

an inline style rule would override the rule of `mango.css` and we should apply `'none'` to cancel `max-width` and `max-height`

**Comment by Leeingnyo** [comment]
As shown in this picture, the position of a right flip panel isn't on expected position in real mode

스크린샷 2022-07-12 오전 10 13 42

it would be fixed giving `width: fit-content;` to the wrapper div (parent of `uk-img`).

**Review by Leeingnyo — CHANGES_REQUESTED** [review]

File: public/js/reader.js [hunk 1]
```diff
@@ -65,6 +66,10 @@ const readerComponent = () => {
 						this.preloadImage(this.items[idx - 1].url);
 					}
 
+					const savedFitType = localStorage.getItem('fitType');
+					if(savedFitType){
+						this.fitType = savedFitType;
```
[the remark this answers is not in the document]
Comment (Leeingnyo) [reply public/js/reader.js:71]: If you don't mind, please fix to `if (savedFitType) {` 😃

File: src/views/reader.html.ecr [hunk 2]
```diff
@@ -40,18 +40,18 @@
           <%- end -%>
         </div>
 
-        <div x-cloak x-show="!loading && mode !== 'continuous'" class="uk-flex uk-flex-middle" style="height:100vh">
+        <div x-cloak x-show="!loading && mode !== 'continuous'" class="uk-flex uk-flex-middle" style="height:${fitType === 'vert' ? '100vh' : ''">
```
Comment (Leeingnyo) [inline src/views/reader.html.ecr:43]: to use JS expression in attribute, add colon in front of attribute name

```
:style="`${fitType === 'vert' ? 'height:100vh' : ''}`"
```

**Review by Leeingnyo — CHANGES_REQUESTED** [review]

**Comment by Hiers** [comment]
Ah, I didn't realise my problem was deleting `max-width` instead of setting it to `none`, thank you.

Your edit made me realise I never explicitly said it, but I was making the fit image feature to be for paged mode only, which is what made sense to me. Do you think this should be available in continuous mode too?

**Review by Leeingnyo — APPROVED** [review]

**Comment by hkalexling** [comment]
Thanks both! I noticed another small issue - the page flip panels are still not positioned correctly. See screenshot below.

DeepinScreenshot_select-area_20220716113915

The right panel should be the right most 1/3 area of the entire screen, and not the right 1/3 of the page. I believe this is caused by the new `position: relative` style in the parent of `uk-img`. It looks like it can be fixed by moving `position: relative` to the `uk-section` tag instead, since it always has 100% width. Do you mind giving it a try? But let me know if I am missing anything!

**Comment by Hiers** [comment]
Nice catch! And your fix did work.

**Comment by hkalexling** [comment]
LGTM, thanks!

**Pull request merged**